aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -71,7 +71,7 @@ void add_clamp_kernel(TensorIterator& iter, const Scalar& alpha_scalar, const Sc
     auto max_vec = Vec256<scalar_t>(max_scalar);
     cpu_kernel_vec(iter,
       [=](scalar_t a, scalar_t b) __ubsan_ignore_undefined__ -> scalar_t {
-        return std::min(max_scalar, std::max(min_scalar, a + alpha * b));
+        return std::min(max_scalar, std::max(min_scalar, static_cast<scalar_t>(a + alpha * b)));
```

Comment:
I don't know anything about the range of values of the variables involved here, but can an overflow occur here?
`add_clamp_kernel` was added after `BFloat16-math.h` was created, so `std::max` & `std::min` for distinct input types were being handled by BFloat16's overloads. Maybe CI checks will make things clear.

Reply:
Hum, it's actually kind of weird that the right hand side is not inferred to be `scalar_t`, a, b and alpha are all scalar_t, what's going on here...

Reply:
Thanks for reviewing this PR!

Yes, that's interesting! I haven't looked into why it was happening yet, but it seems that the compiler was upcasting the type in some cases, such as this one:

```
pytorch/build/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp.DEFAULT.cpp:74:71: 
error: call of overloaded ‘max(const unsigned char&, int)’ is ambiguous
   74 |         return std::min(max_scalar, std::max(min_scalar, a + alpha * b));
      |                                                                       ^
pytorch/aten/src/ATen/Dispatch.h:66:12: note: in definition of macro ‘AT_PRIVATE_CASE_TYPE_USING_HINT’
   66 |     return __VA_ARGS__();                                                        \
      |            ^~~~~~~~~~~
pytorch/aten/src/ATen/Dispatch.h:407:7: note: in expansion of macro ‘AT_PRIVATE_CASE_TYPE’
  407 |       AT_PRIVATE_CASE_TYPE(NAME, at::ScalarType::Byte, uint8_t, __VA_ARGS__)    \
```

Reply:
Hello @ezyang, do you think that the change in this line, viz. `static_cast<scalar_t>(a + alpha * b)` can potentially lead to overflows for dtypes such as `uint8_t` and `int8_t`? I noticed that [`alpha` is always 1](https://github.com/pytorch/pytorch/blob/076961e8b5299395635aae0b863a24f7e1bc33f3/aten/src/ATen/native/native_functions.yaml#L351-L361), but I don't know what values `a` and `b` can have. 

The CI checks don't seem to have a test that might've resulted in an overflow. Thanks!

Reply:
This is expected as arithmetic operators do not accept types smaller than int as arguments, i.e. result of int8+int8 would be int32.

Reply:
@malfet, thanks for the clarification!

I understand that you meant that the conversion to `int` is expected in such cases, 
but apart from that point, please confirm if you also meant one of these two:
1. The values of `a` & `b` would be low enough not to cause an overflow.
           OR
2. Even if overflows occur, the results won't be considered incorrect
eg. For `uint8_t`, if `a` is 128 & `b` is also `128`, then `static_cast<uint8>(128 + 1 * 128)` would be 0.

Thank you!